cmd/release-controller/release.go

```diff
@@ -228,48 +228,61 @@ func isTagEventConditionNotImported(event *imagev1.NamedTagEventList) bool {
 	return false
 }
 
+// find ImagePullSpec will return a fully-resolved pull spec for an image tag
+// when such a thing exists for the local registry. When the source image is
+// known but an import has not yet succeeded, we return the empty string.
 func findImagePullSpec(is *imagev1.ImageStream, name string) string {
 	for i := range is.Status.Tags {
 		tag := &is.Status.Tags[i]
 		if tag.Tag == name {
 			if len(tag.Items) == 0 {
 				if specTag := findSpecTag(is.Spec.Tags, name); specTag != nil {
 					if from := specTag.From; from != nil && from.Kind == "DockerImage" {
-						return from.Name
+						return "" // the ImageStream controller needs to import this, we can check again later
 					}
 				}
-				return ""
+				return "" // the ImageStream controller has not yet imported this, we can check again later
 			}
-			return tag.Items[0].DockerImageReference
+			return fmt.Sprintf("%s@%s", is.Status.PublicDockerImageRepository, tag.Items[0].Image)
```

Comment:
Pretty sure this would break the desired behavior of importing a quay image into a stable stream and serving from there.  We may need a setting per release stream that controls rules (perhaps just for the stable stream).

Reply:
If it's important to directly pull from quay, sure - but the ImageStream import should fire off and the image should be available from this public registry without issue, right?